File: python/pcce10.py
def solution(data, ext, val_ext, sort_by):

# 데이터 타입
    data_type = {
        "code" : 0,
        "date" : 1,
        "maximum" : 2,
        "remain" : 3
        }

    filtered_data = [d for d in data if d[data_type[ext]] < int(val_ext)]
    # filtered_data를 sort_by를 기준으로 다시 오름차순 정렬
    sorted_filtered_data = sorted(filtered_data, key=lambda x: x[data_type[sort_by]])
    return sorted_filtered_data



data = [[1, 20300104, 100, 80], [2, 20300804, 847, 37], [3, 20300401, 10, 8]]
ext = "date"
val_ext = "20300501"
sort_by = "remain"

solution(data,ext,val_ext,sort_by)

File: c++/pcce10.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int s;

bool cmp(const vector<int>&a,const vector<int>&b){
    return a[s] < b[s];
}

vector<vector<int>> solution(vector<vector<int>> data, string ext, int val_ext, string sort_by) {
    vector<vector<int>> answer;
    
    if(ext == "code")
        s = 0;
    else if(ext == "date")
        s = 1;
    else if(ext == "maximum")
        s = 2;
    else if(ext == "remain")
        s = 3;
    cout << s;
    for(int i=0;i<data.size();i++)
    {
        if(data[i][s]<val_ext)
            answer.push_back(data[i]); // 기준에 맞는 데이터 삽입
    }

    // 정렬 기준 설정
    if(sort_by=="code")
        s = 0;
    else if(sort_by == "date")
        s = 1;
    else if(sort_by == "maximum")
        s = 2;
    else if(sort_by == "remain")
        s = 3;

    sort(answer.begin(),answer.end(),cmp);
    return answer;
        
}

